File: Python/Graph/programmers_107.py
import collections

def solution(n, edge):
    answer = 0
    visit = [-1]*(n+1)
    graph = collections.defaultdict(list)
    for a, b in edge:
        graph[a].append(b)
        graph[b].append(a)
            
    que = collections.deque([(1, 0)])
    count = 0
    while que:
        val = que.popleft()
        v = val[0]
        count = val[1]
        if visit[v] == -1:
            visit[v] = count
            count += 1
            for e in graph[v]:
                que.append((e, count))
    
    for v in visit:
        if v == max(visit):
            answer += 1
            
    return answer

File: JavaScript/programmers_107.js
// LV.2 땅따먹기

function solution(land) {
  for (let i = 1; i < land.length; i++) {
    land[i][0] += Math.max(
      land[i - 1][1],
      Math.max(land[i - 1][2], land[i - 1][3])
    );
    land[i][1] += Math.max(
      land[i - 1][0],
      Math.max(land[i - 1][2], land[i - 1][3])
    );
    land[i][2] += Math.max(
      land[i - 1][0],
      Math.max(land[i - 1][1], land[i - 1][3])
    );
    land[i][3] += Math.max(
      land[i - 1][0],
      Math.max(land[i - 1][1], land[i - 1][2])
    );
  }

  return Math.max.apply(null, land[land.length - 1]);
}
